flow-server/src/main/java/com/vaadin/flow/component/Svg.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import org.jsoup.helper.DataUtil;
+
+import com.vaadin.flow.dom.Element;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * A component that displays a given SVG image.
+ * <p>
+ * Note that it is the developer's responsibility to sanitize and remove any
+ * dangerous parts of the SVG before sending it to the user through this
+ * component. Passing raw input data to the user will possibly lead to
+ * cross-site scripting attacks.
+ * <p>
+ * Note, because of implementation details, we currently wrap the SVG in a div
+ * element. This might change in the future.
+ *
+ * @author Vaadin Ltd
+ * @since 24.0
+ */
+public class Svg extends Component {
+
+    /**
+     * Creates an instance based on the given SVG input. The string must have
+     * exactly one root element.
+     *
+     * @param stream
+     *            the SVG to display
+     */
+    public Svg(InputStream stream) {
+        super(new Element("div"));
+        setSvg(stream);
+    }
+
+    /**
+     * Creates an instance based on the given SVG string. The the string must
+     * have exactly one root element.
+     *
+     * @param svg
+     *            the SVG to display
+     */
+    public Svg(String svg) {
+        super(new Element("div"));
+        if (svg == null || svg.isEmpty()) {
+            throw new IllegalArgumentException("SVG cannot be null or empty");
+        }
+        validateAndSet(svg);
+    }
+
+    /**
+     * Creates an empty Svg.
+     */
+    public Svg() {
+        super(new Element("div"));
+    }
+
+    /**
+     * Sets the graphics shown in this component.
+     *
+     * @param svg
+     *            the SVG string
+     */
+    public void setSvg(String svg) {
```

Comment:
I suggest to rename the setters for the content of the Svg component to `setContent`, since having a `svg` variable and calling a `setSvg` method on it is verbose and even a little confusing.
```suggestion
    public void setContent(String svgContent) {
```

Reply:
I don't agree. That method name would confuse people to pass in only content of the top level svg element.

Reply:
In Image we have setSrc. Don't like that either, especially in that shorthand form.

Reply:
I get your concern about a situation that someone might pass children contents of a `svg` instead of including the `<svg ...> </svg>` itself as well, but imagining myself reading/reviewing a code/documentation/tutorial (having a `Svg svg = new Svg();` in the context), I still prefer:
```
svg.setContent("<svg ...></svg>");
```
over
```
svg.setSvg("<svg ...></svg>");
```
Maybe this is only my personal taste.

Btw, we can address that issue by a good documentation. Also, given the fact that the implementation of the `validateAndSet` is taking care of the presence of additional contents such as `<?xml ...>` or `<!DOCTYPE svg... >`, or not having the `<svg ...>` in the input, I don't see a reason to add the verbosity.